app/controllers/welcome_controller.rb

```diff
@@ -0,0 +1,24 @@
+class WelcomeController < ApplicationController
+    before_action :navbar_categories, only: [:index]
+    
+  def index
+    x = rand(1..User.all.length)
```

Comment:
Try not to use single letter variables.  I know naming things is hard but try to describe what it is so your code is easier to read.  If you really can't avoid them at least leave a comment.